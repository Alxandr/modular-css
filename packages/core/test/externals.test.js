"use strict";

const dedent = require("dedent");
const namer  = require("test-utils/namer.js");
const { find, cwd } = require("test-utils/fixtures.js");

const Processor = require("../processor.js");

describe("/processor.js", () => {
    describe("externals", () => {
        var processor;
        
        beforeEach(() => {
            processor = new Processor({
                namer,
                cwd,
            });
        });
        
        it("should fail if not a valid composition reference", async () => {
            await expect(processor.string(
                "./invalid-external.css",
                dedent(`
                    :external(some garbage here) { }
                `)
            )).rejects.toMatchObject({
                message : expect.stringContaining(`SyntaxError: Expected`),
            });
        });

        it("should fail if not referencing another file", async () => {
            await expect(processor.string(
                "./invalid-external.css",
                dedent(`
                    :external(a) { }
                `)
            )).rejects.toMatchObject({
                message : expect.stringContaining(`externals must be from another file`),
            });
        });

        it("should fail on bad class references", async () => {
            await expect(processor.file(find("externals-invalid.css"))).rejects.toMatchObject({
                message : expect.stringContaining(`Invalid external reference: nopenopenope`),
            });
        });
        
        it("should support overriding external values", async () => {
            await processor.file(find("externals.css"));
                
            const result = await processor.output();
            

            expect(result.css).toMatchSnapshot();
        });
    });
});
