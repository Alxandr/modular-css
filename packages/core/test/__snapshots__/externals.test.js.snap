// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/processor.js externals should support overriding external values 1`] = `
"/* fixtures/folder/folder.css */
.folder { margin: 2px; }
/* fixtures/simple.css */
.fooga {
    color: red;
}
/* fixtures/local.css */
.booga { background: green; }
/* fixtures/externals.css */
.a .fooga {
    color: green;
}
.b:hover .booga {
    background: blue;
}
.fooga .booga {
    color: red;
}
"
`;
